spec/system/candidate_interface/decoupled_references/candidate_requests_a_reference_spec.rb

```diff
@@ -176,7 +191,40 @@ def when_i_have_a_cancelled_reference
     @reference = create(:reference, :cancelled, application_form: @application_form)
   end
 
+  def when_i_have_a_failed_reference
+    @reference = create(:reference, :email_bounced, application_form: @application_form)
+  end
+
   def and_i_click_the_resend_reference_link
     click_link 'Send request again'
   end
+
+  def then_i_see_the_references_review_page
+    expect(page).to have_current_path candidate_interface_decoupled_references_review_path
+  end
+
+  def when_i_click_the_retry_request_link
+    click_link 'Retry request'
+  end
+
+  def and_i_continue_with_a_blank_email_address
+    fill_in 'Referee’s email address', with: ''
+    click_button 'Send reference request'
+  end
+
+  def then_i_see_email_address_validation_errors
+    expect(page).to have_content('Enter your referee’s email address')
+  end
+
+  def when_i_change_the_email_address
+    fill_in 'Referee’s email address', with: 'john@example.com'
+  end
+
+  def and_i_confirm_that_i_am_ready_to_retry_a_reference_request
+    click_button 'Send reference request'
+  end
+
+  def and_the_reference_email_address_has_been_updated
+    expect(@reference.reload.email_address).to eq('john@example.com')
```

Comment:
For clarity, maybe we should create the reference in `when_i_have_a_failed_reference` with an explicit email that differs from this one.